lib/actions/jlinkTargetActions.js

```diff
@@ -189,12 +190,29 @@ export function canWrite() {
         // TODO: get the UICR address from the target definition. This value
         // works for nRF51s and nRF52s, but other targets might use a different one!!!
         const appState = getState().app;
-        const targetMemMap = appState.target.memMap;
-        const fileMemMaps = appState.file.memMaps;
-        const { uicrBaseAddr: uicrAddr, uicrSize } = appState.target.deviceInfo;
-        const { isMemLoaded } = appState.target;
+        const {
+            memMap: targetMemMap,
+            deviceInfo,
+            isMemLoaded,
+        } = appState.target;
+        const { isMcuboot } = appState.mcuboot;
+        const {
+            memMaps: fileMemMaps,
+            mcubootFilePath,
+        } = appState.file;
+        const {
+            uicrBaseAddr: uicrAddr,
+            uicrSize,
+        } = deviceInfo;
         let isWritable = false;
```

Comment:
If all calls to `dispatch(targetActions.targetWritableKnownAction(...));` is called with `true` or `false` then `isWritable` can simply be removed, since its not used otherwise.

Reply:
Simplified.